elixir/apps/web/lib/web/live/gateways_live/show.ex

```diff
@@ -1,6 +1,20 @@
 defmodule Web.GatewaysLive.Show do
   use Web, :live_view
 
+  alias Domain.Gateways
+  alias Domain.Resources
+
+  def mount(%{"id" => id} = _params, _session, socket) do
+    {:ok, gateway} = Gateways.fetch_gateway_by_id(id, socket.assigns.subject, preload: :group)
+    {:ok, resources} = Resources.list_resources_for_gateway(gateway, socket.assigns.subject)
+    {:ok, assign(socket, gateway: gateway, resources: resources)}
+  end
+
+  defp time_diff_now(rel_time) do
```

Comment:
Gotcha.  Do you want this component to format and display the entire string?

So using it like so:
```
<.relative_datetime good_arg_name={@gateway.last_seen_at} />
```
Would then produce something like:
```
<span>10 minutes ago</span>
or
<span>1 hour ago</span>
or
<span>In 30 minutes</span>   <-- This would be if you provide a relative time in the future
```

Reply:
Yes, I was thinking about something like this. I think we will have so many dates around that we must have a component like that. Also, this component should have a way for the user to see the actual timestamp (not relative) if the user needs to copy-paste it. And we will eventually make it relative to the user timezone in the future.

Reply:
👍 makes sense.  Will start working on that.